rules/opentitan_test.bzl

```diff
@@ -0,0 +1,374 @@
+        "//sw/host/opentitantool:test_resources",
+        "//hw:verilator",
+    ]
+    required_tags = ["verilator"]
+    kwargs.update(
+        args = required_args + args,
+        data = required_data + data,
+        local = local,
+        otp = otp,
+        rom = rom,
+        tags = required_tags + tags,
+        test_runner = _BASE_PARAMS["test_runner"],
+        timeout = timeout,
+    )
+    return kwargs
+
+def cw310_params(
+        # Base Parameters
+        args = _BASE_PARAMS["args"] + [
+            "--exec=\"console -q -t0\"",
+            "--exec=\"bootstrap $(location {flash})\"",
+            "console",
+            "--exit-failure=FAIL",
+            "--exit-success=PASS",
+            "--timeout=3600",
+        ],
+        data = _BASE_PARAMS["data"],
+        local = _BASE_PARAMS["local"],
+        otp = _BASE_PARAMS["otp"],
+        rom = _BASE_PARAMS["rom"].format("fpga_cw310"),
+        tags = _BASE_PARAMS["tags"] + ["cpu:4"],
+        timeout = _BASE_PARAMS["timeout"],
+        # CW310-specific Parameters
+        # None
+        **kwargs):
+    """A macro to create CW310 parameters for OpenTitan functional tests.
+
+    This macro emits a dictionary of parameters which are pasted into the
+    ChipWhisperer-310 FPGA specific test rule.
+
+    Args:
+        @param args: Extra arguments to pass the test runner `opentitantool`.
+        @param data: Data dependencies of the test.
+        @param local: Whether the test should be run locally without sandboxing.
+        @param otp: The OTP image to use.
+        @param rom: The ROM image to use.
+        @param tags: The test tags to apply to the test rule.
+        @param timeout: The timeout to apply to the test rule.
+    """
+    required_args = [
+        "--rcfile=",
+        "--logging=info",
+        "--interface=cw310",
+        "--conf=sw/host/opentitantool/config/opentitan_cw310.json",
+    ]
+    required_data = [
+        "//sw/host/opentitantool:test_resources",
+    ]
+    required_tags = [
+        "cw310",
+        "exclusive",
+    ]
+    kwargs.update(
+        args = required_args + args,
+        data = required_data + data,
+        local = local,
+        otp = otp,
+        rom = rom,
+        tags = required_tags + tags,
+        timeout = timeout,
+    )
+    return kwargs
+
+def _format_list(param_name, list1, datadict, **kwargs):
+    """Concatenate and format list items.
+
+    This is used to prepare substitutions in user-supplied args to the
+    various test invocations (ie: the location of flash).
+    Args:
+        @param param_name: The name of the item in `datadict`.
+        @param list1: A list of items to prepend to the list item from datadict.
+        @param datadict: A dictionary of per-test parameters.
+        @param **kwargs: Values to pass to the format function.
+    Returns:
+        list[str]
+    """
+    return [x.format(**kwargs) for x in list1 + datadict.pop(param_name, [])]
+
+def _unique_deps(*deplists):
```

Comment:
Isn't this what depset is for...? Maybe I'm missing something.

Reply:
This was there previously, I think @cfrantz put it there to merge the list OTTF deps with user-provided deps and remove duplicates. I just copied it over to the new `opentitan_test.bzl` file. After reading the depset [documentation](https://docs.bazel.build/versions/main/skylark/lib/depset.html#to_list), it seems that calling the `.to_list()` method on a depset will remove any duplicates as well. I created another commit with this change. It seems like it's building successfully. @cfrantz: is there a reason you went with the custom helper function? or were you not aware of the `.to_list()` method on depsets? (Just want to make sure I'm not overlooking something here.)